src/borrow_tracker/stacked/mod.rs

```diff
@@ -1181,7 +989,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 // uncovers a non-supported `extern static`.
                 let alloc_extra = this.get_alloc_extra(alloc_id)?;
                 trace!("Stacked Borrows tag {tag:?} exposed in {alloc_id:?}");
-                alloc_extra.stacked_borrows.as_ref().unwrap().borrow_mut().exposed_tags.insert(tag);
+                alloc_extra.borrow_tracker.expose_tag(tag);
```

Comment:
As I already said last round:
Why does Stacked Borrows call back into the generic borrow tracking machinery here? This should call assume_sb. Then you can probably remove the new expose_tag method on the per-alloc state.